### PR TITLE
Raise error when not enough data when converting text to MDS 

### PR DESCRIPTION
**Pull request by KuuCi**
In response to https://databricks.atlassian.net/browse/MCLOUD-4727, where users were seeing that if they didn't have enough data, no MDS shards would be made, but no error is thrown until much later.

### CODE REVIEW

File: llmfoundry/command_utils/data_prep/convert_text_to_mds.py [hunk 1]
```diff
@@ -468,6 +469,11 @@ def convert_text_to_mds(
             trust_remote_code,
         )
 
+    index_path = os.path.join(local_output_folder, 'index.json')
+    with open(index_path, 'r') as index_file:
+        if not json.load(index_file)['shards']:
+            raise ValueError('Input data was too small, no shards written to output folder.')
```
Comment (dakinggg) [inline llmfoundry/command_utils/data_prep/convert_text_to_mds.py]: @jjanezhang will want this to be a custom error like the other ones i think?

Comment (jjanezhang) [reply llmfoundry/command_utils/data_prep/convert_text_to_mds.py]: Yes! we should add a custom error here telling the user they have too few dataset examples

**Comment by KuuCi** [comment]
vincent-test-FBzlIg for manual test

**Review by jjanezhang — APPROVED** [review]
lgtm, if the input folder is not helpful to surface to ft users, then no need to pass it

File: llmfoundry/utils/exceptions.py [hunk 2]
```diff
@@ -348,6 +349,14 @@ def __init__(self, dataset_name: str, split: str) -> None:
         super().__init__(message, dataset_name=dataset_name, split=split)
 
 
+class DatasetTooSmallError(UserError):
+    """Error thrown when the dataset is too small to be processed."""
+
+    def __init__(self) -> None:
+        message = f'Your dataset is too small to be converted to MDS.'
```
Comment (dakinggg) [inline llmfoundry/utils/exceptions.py:356]:
```suggestion
        message = f'Your dataset is too small and produced no complete samples during preprocessing. Please provide more data.'
```

**Review by dakinggg — APPROVED** [review]

**Pull request merged**